src/target/stm32mp15.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * This file is part of the Black Magic Debug project.
+ *
+ * Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+ * Written by ALTracer <tolstov_den@mail.ru>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file implements STM32MP15 target specific functions for detecting
+ * the device and providing the XML memory map.
+ *
+ * References:
+ * RM0436 - STM32MP157 advanced Arm®-based 32-bit MPUs, Rev. 5
+ *   https://www.st.com/resource/en/reference_manual/rm0436-stm32mp157-advanced-armbased-32bit-mpus-stmicroelectronics.pdf
+ */
+
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+
+/* Access from processor address space.
+ * Access via the debug APB is at 0xe0081000 over AP1. */
+#define DBGMCU_IDCODE      0x50081000U
+#define STM32MP15_UID_BASE 0x5c005234U
+
+#define DBGMCU_IDC (DBGMCU_IDCODE + 0U)
+#define DBGMCU_CR  (DBGMCU_IDCODE + 4U)
```

Comment:
`_CR` -> `_CTRL`? We assume that's what this register is, which is why the "CR" nomenclature is bad as we could be wrong.

Reply:
`_CTRL` was a suffix which Arterytek decided to use for their FLASH controller (#1547), so unrelated. STM32 always used `_CR` for "Control Registers" in most of peripherals' documentation, and would not like to be the one to break the convention. `DBGMCU_CR` allows inhibiting CM4 sleep states and is not part of CoreSight ROM table.

Reply:
We honestly don't really care what Arterytek decided to do as we're trying to accomplish harmonised convention across BMD without having to know insane detail of the manufacturer's foibles and nomenclature to read and understand the different target support, and getting blind-sided each time one switches file with "yet another new convention".

In the rest of the code base the `_CTRL` suffix is used most consistently to refer to a control register. `_CR` doesn't mean anything as it's been used to mean things other than control registers and is too short. The convention we've been establishing in BMD is:

`CHIP_PERIPH_BASE` -> memory base address for a peripheral or memory region
`CHIP_PERIPH_CTRL` -> control register associated with that peripheral
`CHIP_PERIPH_STATUS` -> status register associated with that peripheral

where `CHIP` is the chip suffix (`STM32MP15` here) and `PERIPH` is the name of the peripheral from the datasheet (`DBGMCU` here).

Reply:
Please reopen this on updated patchset contents. I don't quite understand how much level of detail is required here.